### ISSUE TITLE
map_elements gives error with all-null struct elements and a return_dtype

### ISSUE THREAD

**Issue by dkumor** [pola-rs/polars#11862]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
def return_null_struct(row):
    return {"column": None}

pl.DataFrame({"a":[1,2]}).with_columns(col("a").map_elements(return_null_struct,return_dtype=pl.Struct([pl.Field('column',pl.Float64)])).alias("mapped")).schema
```

```
SchemaError: expected output type 'Struct([Field { name: "column", dtype: Float64 }])', got 'Struct([Field { name: "column", dtype: Null }])'; set `return_dtype` to the proper datatype
```

### Issue description

When returning a struct from a function with map_elements, if all the outputs of a field happen to be null then the inferred struct schema for that field will be null. An all-null column output should be compatible with any nullable dtype field.

Using automatic type inference in this case would disallow `pl.concat` and the likes on output results, since the schema would no longer match the schema of other dataframes. It also causes a crash when using `col().struct.field().str....` functions since the value is not considered a null string.

### Expected behavior

The behavior should match returning a single value:
```python
def return_null(row):
    return None

pl.DataFrame({"a":[1,2]}).with_columns(col("a").map_elements(return_null,return_dtype=pl.Float64).alias("mapped")).schema
```

```
{'a': Int64, 'mapped': Float64}
```

### Installed versions

<details>

```
Replace this line with the output of pl.show_versions(). Leave the backticks in place.
```

</details>

**Comment by dkumor** [comment]
The same issue happens if you return python ints in the struct column, but the schema says they are floats. Would it be possible to have the same type-conversion rules on struct/list/list of struct as is currently done when constructing DataFrames?

**Comment by dkumor** [comment]
I think this is related to #11878 and #11849, with the main difference being that the source of the series is the map_elements function rather than an explicit constructor.

**Comment by john-sungjin** [comment]
also having this issue! confirmed on polars 0.20.7.

**Comment by Baukebrenninkmeijer** [comment]
I'm having this issue where the target column is null, and my function returns a default dictionary, but when the whole column is null, the returned series has length 0. This looks like the same issue. I'm running 1.2.1.

**Comment by cmdlineluser** [comment]
@Baukebrenninkmeijer Perhaps you can make a new issue?

This particular issue no longer errors, which suggests yours is different:

```python
def return_null_struct(row):
    return {"column": None}

(pl.DataFrame({"a":[1,2]})
   .with_columns(pl.col("a").map_elements(return_null_struct,return_dtype=pl.Struct([pl.Field('column',pl.Float64)])).alias("mapped")).schema
)
# Schema([('a', Int64), ('mapped', Struct({'column': Float64}))])
```

**Comment by Baukebrenninkmeijer** [comment]
@cmdlineluser Yes, just realized. Mine is fixed by setting `skip_nulls=False`. However, the fact that this returns an empty series is still unexpected behaviour imo. I'll create a new ticket for this.

**Comment by dkumor** [comment]
It looks like this particular issue has been fixed! Closing.
